examples/tpm-direct/values.yaml

```diff
@@ -0,0 +1,15 @@
+spire-agent:
+  nodeAttestor:
+    k8sPsat:
+      enabled: false
+    tpmDirect:
+      enabled: true
+
+spire-server:
+  controllerManager:
+    # K8s labels have a 63 character limit. TPM hashes are 64 chars. So you need to label the node with two labels with half of the tpm's hash each. The 'node-restriction.kubernetes.io/' prefix is so that the
+    # nodes can't update the hash themselves, an important security constraint.
+    parentIDTemplate: 'spiffe://{{ .TrustDomain }}/spire/agent/tpm/{{ index .NodeMeta.Labels "node-restriction.kubernetes.io/tpm-pubhash" }}{{ index .NodeMeta.Labels "node-restriction.kubernetes.io/tpm-pubhash2" }}'
```

Comment:
Should the chart check this `restriction.kubernetes.io` is part of the template as some sort of safetynet?

Reply:
I wish we could, but it is a restriction on the labels being manually added to the node by the end user. Its not done in the chart, so not something we can check in the chart.

k8s itself does enforce the restriction.If you try and load it all in one string, it complains that the label is too long.

Once the tpm plugin pr's merge, I'm going to provide documentation at https://github.com/spiffe/spiffe.io/pull/290

Reply:
Yes outside of the chart an enduser could do that indeed. But this value specifically we can control it right?

Reply:
Oh. I think I may have been misunderstanding what you've been saying...  I thought you were talking about the character limit, not the parentIDTemplate itself.

That would be good to do if we could figure out a way to do it. I can't think of a way off hand to be able to do that, outside of writing our own gotl parser?

Reply:
@marcofranssen Are you asking just to check if the string `restriction.kubernetes.io` is present?

Reply:
it was about the char limit indeed, but seeing Kevins response that is not possible prior to parsing the template I guess. So no worries, I'm fine merging like this. Just tried to see if we could failsafe the user.